### PR TITLE
Add missing fields to AddressUtxo

### PR DESCRIPTION
**Pull request by bentaye**
Add `dataHash`, `inlineDatum`, `referenceScriptHash` fields as decribed [here](https://github.com/blockfrost/openapi/blob/a025b0a994eaea80dcd1a575f54704d7aaffc9da/openapi.yaml#L6950)

### CODE REVIEW

File: src/main/java/io/blockfrost/sdk/api/model/AddressUtxo.java [hunk 1]
```diff
@@ -31,11 +31,22 @@ public class AddressUtxo {
     /**
      * Array of TransactionOutputAmount objects
      */
-    private List<TransactionOutputAmount> amount = new ArrayList<TransactionOutputAmount>();
+    private List<TransactionOutputAmount> amount = new ArrayList<>();
     /**
      * Block number of the UTXO
```
Comment (bentaye) [inline src/main/java/io/blockfrost/sdk/api/model/AddressUtxo.java:36]: The doc says `Block hash of the UTXO` for the description of that field, not `number`
I an change that as part of this PR if required

**Comment by bentaye** [comment]
Testing wise, the only relevant test I can see is `AddressServiceTests.addressUtxos_willReturn_allAddressUtxos`
The address that is used has 3 UTXOs, all have null values for  `dataHash`, `inlineDatum`, `referenceScriptHash`

Where an I find an address that has values for those fields? 
Or should this address have those fields, and there is something missing in the PR?

**Comment by satran004** [comment]
> Testing wise, the only relevant test I can see is `AddressServiceTests.addressUtxos_willReturn_allAddressUtxos` The address that is used has 3 UTXOs, all have null values for `dataHash`, `inlineDatum`, `referenceScriptHash`
> 
> Where an I find an address that has values for those fields? Or should this address have those fields, and there is something missing in the PR?

@bentaye I executed a transaction to create an utxo with inline datum and reference script hash at the address used in the above test. You should now see the following utxo with required data (on testnet)

Address: addr_test1qqy07jaue8tru20877ak7wxpuagrqqpm2pdacfjjtv4z3elcn8dnk52656jspgq03ts2sl6jvefwakdacwfy605m9ydselehdg
Utxo: 1aaf4d00fcf090a4c39718c0f8a49bf33195a205dc868d59631dfe6eba055c30#2
dataHash = 09360dd77aa5a7890addd66d6bad404570d76ec99918539ddf81b600a70fc4c1
Inline_dataum = 1907d0
referenceScriptHash = 3a888d65f16790950a72daee1f63aa05add6d268434107cfa5b67712

Can you please verify and update the test ?

**Comment by bentaye** [comment]
I added assertions to the test to check that for each of the new field, there is at least one Utxo in the list with a non null value for that field

**Comment by satran004** [comment]
@bentaye Thanks a lot. Looks good to me.

**Review by satran004 — APPROVED** [review]

**Comment by satran004** [comment]
Merging this PR. The CI integration tests are failing for other unrelated tests. 
So need to fix CI issue before next release.

**Pull request merged**